packages/rich-text/src/apply-format.js

```diff
@@ -52,14 +52,10 @@ export function applyFormat(
 		// with the format applied.
 		} else {
 			const previousFormat = newFormats[ startIndex - 1 ] || [];
-			const hasType = find( previousFormat, { type: format.type } );
 
 			return {
-				formats,
-				text,
-				start,
-				end,
-				formatPlaceholder: hasType ? undefined : format,
+				...value,
```

Comment:
In the equivalent function in the native implementation, we don't spread to preserve the other properties. Namely, `replacements` would be dropped. I don't think that's expected?

https://github.com/WordPress/gutenberg/blob/2889c29458489876a941cf4aaae093b10c1990e8/packages/rich-text/src/apply-format.native.js#L42-L51

https://github.com/WordPress/gutenberg/blob/2889c29458489876a941cf4aaae093b10c1990e8/packages/rich-text/src/apply-format.native.js#L60

Reply:
I'm finding similar issues in `remove-format.native.js`

Reply:
I did not realise there was a `native` version, I think it was added recently. I have no idea why these files differ. There's no comment about it. No we have to maintain two files that do the same thing? Cc @hypest

Reply:
It was added in #12249. Cc @koke

Reply:
Fixed in 62fa458.

Reply:
Updated everything for mobile, and was approved in https://github.com/WordPress/gutenberg/pull/13948#pullrequestreview-213902940. Thanks for catching this.